### PR TITLE
Docs for Renewable Energy Models

### PR DESCRIPTION
**Pull request by dguittet**
## Addresses issue:
July internal release: https://github.com/gmlc-dispatches/dispatches/issues/45

## Changes proposed in this PR:
RST files for PEM electrolyzer, Battery, Wind & Electricity Splitter

### CODE REVIEW

File: docs/models/renewables/wind_turbine.rst [hunk 1]
```diff
@@ -1,37 +1,45 @@
...
+The DISPATCHES Wind Turbine Model calculates the electricity production from a wind farm using wind resource data and
+the PySAM Windpower module. The model assumes a single turbine with a powercurve from the ATB Turbine 2018 Market Average.
+Wake effects are not modeled, and PySAM's default losses are assumed. The power output of the entire farm of any size is
+calculated by scaling the output from the single turbine.
 
 Degrees of Freedom
 ------------------
 
-The Wind Turbine Model has X degrees of freedom.
+The Wind Turbine Model has 0 degrees of freedom.
```
Comment (andrewlee94) [inline docs/models/renewables/wind_turbine.rst:12]: In this case, it is probably worth noting any parameters that need to be provided for PySAM.

**Review by andrewlee94 — APPROVED** [review]

**Comment by andrewlee94** [comment]
@ksbeattie Any idea why the readthedocs build failed?

**Comment by lbianchi-lbl** [comment]
> @ksbeattie Any idea why the readthedocs build failed?

From the [build log](https://readthedocs.org/projects/dispatches/builds/14209694/), it seems that there's one duplicate entry warning for one of the Python objects:

```txt
/home/docs/checkouts/readthedocs.org/user_builds/dispatches/checkouts/48/docs/models/renewables/wind_turbine.rst:44: WARNING: duplicate object description of dispatches.models.renewables_case.unit_model.wind_turbine, other instance in models/renewables/pem_electrolyzer, use :noindex: for one of them
```

File: docs/models/renewables/wind_turbine.rst (hunk 1 above)
Comment (dguittet) [reply docs/models/renewables/wind_turbine.rst]: Added a sentence to Model Structure section

**Review by andrewlee94 — DISMISSED** [review]

**Review by ksbeattie — APPROVED** [review]
Hopefully the .readthdocs.yaml change we talked about today will allow the autodoc stuff to work.

**Comment by dguittet** [comment]
@lbianchi-lbl @ksbeattie I added the updated readthedocs.yml but it seems like the readthedocs build is failing with the autodocs:

```
WARNING: autodoc: failed to import class 'models.renewables_case.battery.BatteryStorage' from module 'dispatches'; the following exception was raised:
No module named 'dispatches'
WARNING: autodoc: failed to import class 'models.renewables_case.elec_splitter.ElectricalSplitter' from module 'dispatches'; the following exception was raised:
No module named 'dispatches'
WARNING: autodoc: failed to import class 'models.renewables_case.pem_electrolyzer.PEM_Electrolyzer' from module 'dispatches'; the following exception was raised:
No module named 'dispatches'
WARNING: autodoc: failed to import class 'models.renewables_case.wind_power.Wind_Power' from module 'dispatches'; the following exception was raised:
No module named 'dispatches'
```

Note when I run this locally, on my Python environment where dispatches is installed, the sphinx build works.

Issue is that the build environment of readthedocs is missing `dispatches`. I think we need either a `python setup.py install` step in order to install `dispatches` from scratch; or use https://github.com/readthedocs/sphinx-autoapi which seems to be developed for this use case.

**Comment by lbianchi-lbl** [comment]
@dguittet I think the reason why this is not working is that incorporating the changes to `.readthedocs.yml` from #49 only is not enough on its own, since in the current version `requirements-dev.txt` does not install `dispatches`. I overlooked this detail during the meeting today; sorry about that.

You could try adding the line `--editable .` to `requirements-dev.txt` (i.e. incorporating the changes to `requirements-dev.txt` in #49), which should be enough to install `dispatches` in the RTD build environment.

**Comment by dguittet** [comment]
@lbianchi-lbl Thanks! I pulled in requirements-dev.txt and setup.py from your PR and now it builds.

**Review by ksbeattie — APPROVED** [review]
LGTM

**Review by andrewlee94 — APPROVED** [review]

**Pull request merged**